### ISSUE TITLE
E3503 does not match CloudFormation - requires ValidationDomain when CF does not want it

### ISSUE THREAD

**Issue by mrapczynski** [aws-cloudformation/cfn-python-lint#1619]
*cfn-lint version: (`cfn-lint --version`)*

cfn-lint 0.33.2

*Description of issue.*

I created an ACM certificate resource, and there were problems configuring the `DomainValidationOptions` block. If using DNS validation, the only properties needed are `DomainName` and `HostedZoneId`. However, cfn-lint was demanding a third property named `ValidationDomain`. When submitting the stack for deployment to CF, it triggered an immediate rollback because CF views `HostedZoneId` and `ValidationDomain` as mutually exclusive.

Adding an ignore rule to skip the E3503 error allowed me to proceed without issues. This rule should be adjusted to match what CF enforces.

**Sample:**

```yaml
Resources:
  Certificate:
    Type: AWS::CertificateManager::Certificate
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
            - E3503
    Properties:
      DomainName: "*.aws.domain.com"
      ValidationMethod: DNS
      DomainValidationOptions:
        - DomainName: aws.domain.com
          HostedZoneId: !ImportValue SubdomainHostedZoneId
```

**Comment by kddejong** [comment]
Agreed.  Thanks for reporting.  We will work on getting this fixed.
